koswat/cost_report/infrastructure/multi_infrastructure_profile_costs_calculator.py

```diff
@@ -16,15 +16,49 @@
 
 @dataclass
 class MultiInfrastructureProfileCostsCalculator:
-    infrastructure_calculators: dict[str, InfrastructureCostsCalculator]
+    """
+    Calculator that contains all possible "infrastructure" calculators
+    (`InfrastructureProfileCostsCalculator`) one for each of the available
+    infrastructures in the current `KoswatScenario`.
+    Its `calculate` method only requires a reinforcement
+    (`ReinforcementProfileProtocol`) to determine all infrastructures' costs.
+    """
+
+    infrastructure_calculators: list[InfrastructureProfileCostsCalculator] = field(
+        default_factory=lambda: []
+    )
```

Comment:
```suggestion
    infrastructure_calculators: list[InfrastructureProfileCostsCalculator] = field(
        default_factory=list
    )
```

Reply:
Your approach has actually given me problems in the past days, I do not recall exactly how but I remember I could not fix it, so that's why I'm using the lambda expression instead. Won't commit.